### PR TITLE
New pipe bombs

### PR DESCRIPTION
**Pull request by Brian-Otten**
#### Summary

 ```SUMMARY: Content "Adds more craftable pipe bombs'"```

#### Purpose of change
Crafting of explosives is a bit of a mess with a ton of options using specific ingredients, the goal is to genericize the recipes and use the new infrastructure that's in place. this should make crafting them in game less of a hassle, and contributing new explosives in the future should be much easier for devs.

#### Describe the solution
Adds primitive, improvised, and bootleg pipebombs using new explosives infrastructure.
These are made using gunpowder, AN (and ilk), and military explosives respectively.
Obsoletes pipe bombs.
Replaces pipe bomb spawns with improvised pipe bombs

#### Testing
Load up a world and spawn some pipe bombs and active makeshift pipe bombs.
Merge this PR
No crashes/bugs from the now obsolete pipe bombs.
Craft all 3 new pipe bombs.
pipe bombs all work and explode as intended.

#### Additional context
image

image

### CODE REVIEW

**Comment by Brian-Otten** [comment]
Made a mistake in an attempt to rebase and it requested a bunch of reviews, my apologies for that. i think the only intended one was Erk because i replaced the pipe bomb with the new variant in the traders stock.

**Comment by mkikt4743** [comment]
I don't understand removal of the 'normal' pipe bomb. Are they not made by your average dude? I feel like one should be called as it was, so that it seems higher quality

**Comment by Brian-Otten** [comment]
The normal "pipe bomb" was the pipe bomb made with gunpowder, which is essentially the shittiest pipe bomb possible. The primitive pipe bomb is close to the same thing the old pipe bomb used to be. 
The world spawn pipe bombs have been replaced with the "improvised pipe bomb" because that's the one people could make in decent amounts without having to destroy tons of ammunition.

**Comment by mkikt4743** [comment]
ah ok, just sayin improvised makes it seem less powerful than the original

File: data/json/items/tool/explosives.json [hunk 1]
```diff
@@ -1015,12 +1015,13 @@
...
-    "id": "pipebomb",
+    "id": "bootleg_pipebomb",
     "type": "TOOL",
     "category": "weapons",
-    "name": { "str": "pipe bomb" },
-    "description": "This is a section of a pipe filled with explosive materials.  Use this item to light the fuse, which gives you five turns to get away from it before it detonates.  You'll need a lighter or some matches to use it.",
-    "weight": "730 g",
+    "name": { "str": "bootleg pipe bomb" },
+    "looks_like": "pipebomb",
+    "description": "This is a section of a pipe filled with military explosives.  Use this item to pull the pin, which gives you five turns to get away from it before it detonates.",
```
Comment (Amoebka) [inline data/json/items/tool/explosives.json:1023]:
```suggestion
    "description": "This is a section of a pipe filled with military explosives.  Use this item to pull the pin, which gives you six seconds to get away from it before it detonates.",
```
"Turns" are both gamey and generally incorrect. (player's speed isn't constant) Also, the screenshots show 6 seconds.

Comment (Brian-Otten) [reply data/json/items/tool/explosives.json]: Absolutely right, changed them all to six seconds.

**Comment by NoahGoldFox** [comment]
Pipe bombs have always been a classic, ever since the first terminator movie. These would be a pretty fun addition

**Comment by Brian-Otten** [comment]
The last update from master and pr was to add improvised_pipebombs to the stock of the arsonist npc after her quest. (instead of the obsoleted pipebombs.)
This PR should still be good to merge.

**Pull request merged**